src/bootloaders/systemd-class.c

```diff
@@ -246,7 +246,14 @@ bool sd_class_install_kernel(const BootManager *manager, const Kernel *kernel)
                                          "initrd %s/%s\n",
                                          get_kernel_destination_impl(manager),
                                          kernel->target.initrd_path);
+        } else {
```

Comment:
I am having an issue with this `else`. In the scenario where "freestanding" initrd is used for dm-crypt (or dm-verity) setup, if I add an kernel-specific `initrd` it may render the OS unusable, because kernel-specific initrd will be used in the bootloader entry and it will not be able to create and mount its encrypted partitions, including possibly root.

At the very least, we should prefer the global initrd here and issue a warning when kernel-specific initrd is present, but ignored.

Reply:
Ok, I can change that easy

Reply:
changed it